### PR TITLE
[dashboard] Setup (component) integration tests

### PR DESCRIPTION
**Pull request by geropl**
## Description
Makes `yarn test` executed unit (craco test) and (dashboard) integration tests (cypress) in non-interactive mode.
Also, `yarn test` now runs during builds.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by jankeromnes** [comment]
Thanks @geropl! Happy to take a look at this. 👀

> [server] Fix case where we might close ws connections too frequently

Could you please briefly explain why this change was needed? (At first glance it looks unintentional, but I guess this is to fix some test failures?)

Anyway, taking this for a spin. 📀

**Comment by geropl** [comment]
> Could you please briefly explain why this change was needed?

@jankeromnes Sorry, this was a mistake! Somehow got my PRs merged... :see_no_evil: Will pop.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7807#issuecomment-1021075173" title="Approved">geropl</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7807#issuecomment-1021075173" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/package.json
Comment (jankeromnes) [inline components/dashboard/package.json:52]: Thanks for adding this helpful script! 🙏

**Suggestion**: I feel like it would be cool to add it as a `pre` task, so that `yarn test:integration:run` can also work out of the box. What do you think of something like this?

```suggestion
    "test": "yarn test:unit && yarn test:integration:run",
    "pretest:integration:run": "./scripts/run-integration-tests.sh",
```

Note: This implies removing the `yarn test:integration:run` from the bottom of the script.

**Review by jankeromnes — APPROVED** [review]
Looks good to me, thanks!

I was able to successfully run `yarn test` manually. ✨

However, checking the Werft logs, I couldn't find proof that the tests also ran in Werft. Maybe because the dashboard build result is cached, and there were no changes detected to the dashboard code? (I guess this means the tests will only run in Werft once this PR is merged, and the next time someone makes a change to the dashboard code?)

Also, added a suggestion in-line. Please feel free to accept or reject it and then merge. 🚢

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: ecdbb9e23a6650204c6df324ef3084ea770547bb</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7807#issuecomment-1021075173" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7807#pullrequestreview-865898145" title="Approved">jankeromnes</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7807#issuecomment-1021075173" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [geropl,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/package.json
Comment (geropl) [reply components/dashboard/package.json:52]: I choose the current hierarchy so we can run `yarn test:integration:run` separately, against a dev server in watch mode. The script is just a crutch for the parallelims (dev server x cypress). Let's try this a bit, and adjust when it confuses us.

Comment (jankeromnes) [reply components/dashboard/package.json:52]: Makes sense to me, thanks! 🚀

**Pull request merged**
